devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java

```diff
@@ -31,57 +33,96 @@ public AddExtensions(final ProjectWriter writer, final String pom) throws IOExce
         this.pom = pom;
     }
 
-    public boolean addExtensions(final Set<String> extensions) throws IOException {
+    /**
+     * Selection algorithm.
+     *
+     * @param query the query
+     * @param extensions the extension list
+     * @return the list of matching candidates and whether or not a match has been found.
+     */
+    static SelectionResult select(String query, List<Extension> extensions) {
+        String q = query.trim().toLowerCase();
+
+        // Try exact matches
+        Set<Extension> matchesNameOrArtifactId = extensions.stream().filter(extension -> extension.getName().equalsIgnoreCase(q)
+                || matchesArtifactId(extension.getArtifactId(), q)).collect(Collectors.toSet());
+
+        if (matchesNameOrArtifactId.size() == 1) {
+            return new SelectionResult(matchesNameOrArtifactId, true);
+        }
+
+        // Try short names
+        Set<Extension> matchesShortName = extensions.stream().filter(extension -> matchesShortName(extension, q))
+                .collect(Collectors.toSet());
+
+        if (matchesShortName.size() == 1) {
+            return new SelectionResult(matchesShortName, true);
+        }
+
+        // find by labels
+        List<Extension> matchesLabels = extensions.stream()
+                .filter(extension -> extension.labels().contains(q)).collect(Collectors.toList());
+
+        Set<Extension> candidates = new LinkedHashSet<>();
```

Comment:
The old algorithm had a match on partial artifactId and partial name match. It seems to be lost in this new approach.

e.g. `orm-pana` => `hibernate-orm-panache`, `jdbc-postgre` -> JDBC PostgreSQL driver

Reply:
I've re-introduced this behavior.

Reply:
(with tests of course).